test/select_io_dset.c

```diff
@@ -184,6 +200,13 @@ test_no_type_conv(hid_t fid, unsigned chunked, unsigned dtrans, unsigned mwbuf)
     if (H5Dwrite(did, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, dxpl, wbuf) < 0)
         FAIL_STACK_ERROR;
 
+    if (H5Pget_no_selection_io_cause(dxpl, &no_selection_io_cause) < 0)
+        FAIL_STACK_ERROR;
+
+    if (!no_selection_io_cause)
```

Comment:
Why do you ca;ll H5Pget_no_selection_io_cause() here (and in similar places)? Why should it be scalar if there's no cause?

Reply:
Also does check-vfd run with the MPIO vfd (even for serial tests like this)? If so this might fail since it'll use vector I/O.